tests/python/unittest/test_runtime_module_export.py

```diff
@@ -0,0 +1,208 @@
+
+        A = tvm.placeholder((1024,), name='A')
+        B = tvm.compute(A.shape, lambda *i: A(*i) + 1.0, name='B')
+        s = tvm.create_schedule(B.op)
+        f = tvm.build(s, [A, B], "llvm", name="myadd")
+        from tvm.contrib import util
+        temp = util.tempdir()
+        if obj_format == ".so":
+            file_name = "deploy_lib.so"
+        else:
+            assert obj_format == ".tar"
+            file_name = "deploy_lib.tar"
+        path_lib = temp.relpath(file_name)
+        resnet18_cpu_lib.import_module(f)
+        resnet18_cpu_lib.export_library(path_lib)
+        loaded_lib = tvm.module.load(path_lib)
+        assert loaded_lib.type_key == "library"
+        assert loaded_lib.imported_modules[0].type_key == "library"
+
+    def verify_json_import_dso(obj_format):
+        for device in ["llvm"]:
+            if not tvm.module.enabled(device):
+                print("skip because %s is not enabled..." % device)
+                return
+
+        # Get subgraph Json.
+        subgraph_json = ("json_rt_0\n" +
+                         "input 0 10 10\n" +
+                         "input 1 10 10\n" +
+                         "input 2 10 10\n" +
+                         "input 3 10 10\n" +
+                         "add 4 inputs: 0 1 shape: 10 10\n" +
+                         "sub 5 inputs: 4 2 shape: 10 10\n" +
+                         "mul 6 inputs: 5 3 shape: 10 10\n" +
+                         "json_rt_1\n" +
+                         "input 0 10 10\n" +
+                         "input 1 10 10\n" +
+                         "input 2 10 10\n" +
+                         "input 3 10 10\n" +
+                         "add 4 inputs: 0 1 shape: 10 10\n" +
+                         "sub 5 inputs: 4 2 shape: 10 10\n" +
+                         "mul 6 inputs: 5 3 shape: 10 10")
+
+        from tvm.contrib import util
+        temp = util.tempdir()
+        subgraph_path = temp.relpath('subgraph.examplejson')
+        with open(subgraph_path, 'w') as f:
+            f.write(subgraph_json)
+
+        # Get Json and module.
+        A = tvm.placeholder((1024,), name='A')
+        B = tvm.compute(A.shape, lambda *i: A(*i) + 1.0, name='B')
+        s = tvm.create_schedule(B.op)
+        f = tvm.build(s, [A, B], "llvm", name="myadd")
+        try:
+            ext_lib = tvm.module.load(subgraph_path, "examplejson")
+        except:
+            print("skip because Loader of examplejson is not presented")
+            return
+        ext_lib.import_module(f)
+        if obj_format == ".so":
+            file_name = "deploy_lib.so"
+        else:
+            assert obj_format == ".tar"
+            file_name = "deploy_lib.tar"
+        path_lib = temp.relpath(file_name)
+        ext_lib.export_library(path_lib)
+        lib = tvm.module.load(path_lib)
+        assert lib.type_key == "examplejson"
+        assert lib.imported_modules[0].type_key == "library"
+
+    def verify_multi_c_mod_export():
```

Comment:
can we test the case where there are two c modules? so far seems there is only one, we can build another one for sub

Reply:
Here has two c modules. Notice that ` f = tvm.build(s, [A, B], "c", name="myadd")`.  Its target is c, not llvm. I create engine module (c module) is to prove other c modules could work well too even they need options like c++11 / include extra header dirs path and so on.

Reply:
If you would like to c module's submodule, New code could change to this: llvm <- engine module (c module) <- f (c module). That is to say, `resnet18_cpu_lib` only import `engine module (c module)`, but `engine module` import `f (c module). Currently, `resnet18_cpu_lib` import `engine module (c module)` and `f (c module)`.

Reply:
i didn't realize that the engine module is already a c module! then we are fine here

Reply:
Ok. I keep current way.